### PR TITLE
[SYCL] Add a test for generated device code for the group_load_store extension

### PR DESCRIPTION
**Pull request by aelovikov-intel**
Only loads for now. Also adds `detail::naive` property to force usage of the unoptimized implementation, both for testing purposes and potential future experiments/debugging.

### CODE REVIEW

**Comment by aelovikov-intel** [comment]
Next PR built on top of this one: https://github.com/intel/llvm/pull/13673

File: sycl/test/check_device_code/group_load.cpp [hunk 1]
```diff
@@ -0,0 +1,605 @@
...
+template SYCL_EXTERNAL void sycl::ext::oneapi::experimental::group_load<
+    sycl::sub_group, plain_global_ptr<int>, int, opt_blocked>(
+    sycl::sub_group, plain_global_ptr<int>, int &, opt_blocked);
+// CHECK-LABEL: define weak_odr dso_local spir_func void @_ZN4sycl3_V13ext6oneapi12experimental10group_loadINS0_9sub_groupEPU3AS1iiNS3_10propertiesISt5tupleIJNS3_14property_valueINS3_18data_placement_keyEJSt17integral_constantIiLi0EEEEENSA_INS3_21contiguous_memory_keyEJEEENSA_INS3_14full_group_keyEJEEEEEEEEENSt9enable_ifIXaasr6detailE17verify_load_typesIT0_T1_Esr6detailE18is_generic_group_vIT_EEvE4typeESO_SM_RSN_T2_(
+// CHECK-SAME: ptr noundef byval(%"struct.sycl::_V1::sub_group") align 1 [[G:%.*]], ptr addrspace(1) noundef [[IN_PTR:%.*]], ptr addrspace(4) noundef align 4 dereferenceable(4) [[OUT:%.*]], ptr noundef byval(%"class.sycl::_V1::ext::oneapi::experimental::properties.0") align 1 [[PROPERTIES:%.*]]) local_unnamed_addr #[[ATTR0]] comdat !srcloc [[META5]] !sycl_fixed_targets [[META6]] {
+// CHECK-NEXT:  entry:
+// CHECK-NEXT:    tail call spir_func void @_Z22__spirv_ControlBarrierjjj(i32 noundef 3, i32 noundef 3, i32 noundef 912) #[[ATTR2]]
```
Comment (uditagarwal97) [inline sycl/test/check_device_code/group_load.cpp:57]: The device code seems same for naive_blocked and opt_blocked. Reading the comment "Check that optimized implementation is selected.", I'm not sure if that's intended.

**Comment by uditagarwal97** [comment]
Also, why was the lint disabled in this PR? Was clang-format altering the auto-generated CHECKs in the test case?

File: sycl/test/check_device_code/group_load.cpp (hunk 1 above)
Comment (aelovikov-intel) [reply sycl/test/check_device_code/group_load.cpp:57]: That is intended for this PR. The test is supposed to be modified with each functional PR implementing said "optimizations" support.

**Comment by aelovikov-intel** [comment]
> Also, why was the lint disabled in this PR? Is clang-format altering the auto-generated CHECKs in the test case?

It complains about the first auto-generated line describing how to generate the CHECKs. Normally clang's tests have clang-format disabled so long line generated by the script is considered OK.

File: sycl/test/check_device_code/group_load.cpp (hunk 1 above)
Comment (uditagarwal97) [reply sycl/test/check_device_code/group_load.cpp:57]: Ahh.. I see. Makes sense.

**Review by uditagarwal97 — APPROVED** [review]
Changes LGTM!

**Comment by aelovikov-intel** [comment]
CUDA E2E tests failures are unrelated, CUDA AWS pre-commit workflow has been just re-enabled after being broken for weeks.

**Pull request merged**